include/nbl/video/IGPUCommandPool.h

```diff
@@ -344,7 +344,10 @@ class IGPUCommandPool : public core::IReferenceCounted, public IBackendObject
         }
 
         CCommandSegment* m_head = nullptr;
-        core::CMemoryPool<core::PoolAddressAllocator<uint32_t>, core::default_aligned_allocator, false, uint32_t> m_pool;
+        
+        template <typename T>
+        using pool_alignment = core::aligned_allocator<T,COMMAND_SEGMENT_ALIGNMENT>;
+        core::CMemoryPool<core::PoolAddressAllocator<uint32_t>, pool_alignment, false, uint32_t> m_pool;
```

Comment:
can you explain this change?

Reply:
sets the alignment to 64.

Reply:
but shouldn't the `m_pool.emplace` take into account the alignment of what its emplacing?
https://github.com/Devsh-Graphics-Programming/Nabla/blob/6a17dc80e50be0fee6114e2efd2408fbb223968a/include/nbl/core/containers/CMemoryPool.h#LL43C19-L43C20

Reply:
the default alignment is 16
0x7fffddfff010 is a multiple of 16 but not 64

Reply:
> but shouldn't the `m_pool.emplace` take into account the alignment of what its emplacing? https://github.com/Devsh-Graphics-Programming/Nabla/blob/6a17dc80e50be0fee6114e2efd2408fbb223968a/include/nbl/core/containers/CMemoryPool.h#LL43C19-L43C20

the memory where its getting allocated is aligned to 16.

Reply:
```cpp
    class alignas(COMMAND_ALIGNMENT) ICommand
    {
```

you can clearly see it being aligned.

> the memory where its getting allocated is aligned to 16.

this is a problem that needs fixing, 

are you getting 64 here and if not why not?
https://github.com/Devsh-Graphics-Programming/Nabla/blob/6a17dc80e50be0fee6114e2efd2408fbb223968a/include/nbl/core/containers/CMemoryPool.h#LL43C19-L43C20

Reply:
when you malloc memory there isn't any guarantee that it will be on a 64 bit boundary? it doesn't know anything abotu the memory requriments for the struct right?

![image](https://user-images.githubusercontent.com/854359/226191134-bc69200e-41ca-4cc7-bdd0-677aa5e2e2d7.png)

Reply:
> when you malloc memory there isn't any guarantee that it will be on a 64 bit boundary? 

`malloc` doesn't, but `core::aligned_allocator::allocate` asks for the alignment explicitly.

I'm asking you why isn't `CMemoryPool::allocate` being called with `a=64` when emplacing the `CommandSegment` ?

Reply:
you can look at aligned_allocator i think there are two methods allocate and allocate with an alignment. memory pool calls allocated so it defaults to what you pass into the second template parameter. in this case I've passed in COMMAND_SEGMENT_ALIGNMENT which is 64.

Reply:
See whether this function is being called with alignment=64
https://github.com/Devsh-Graphics-Programming/Nabla/blob/c6d0dde3f7b5e4bde019290562997c8dfad28268/include/nbl/core/alloc/SimpleBlockBasedAllocator.h#L104

if it is then the only reason why it would fail is because the Block itself is not aligned to 64
https://github.com/Devsh-Graphics-Programming/Nabla/blob/c6d0dde3f7b5e4bde019290562997c8dfad28268/include/nbl/core/alloc/SimpleBlockBasedAllocator.h#L45

which would require `meta_alignment<64`  here
https://github.com/Devsh-Graphics-Programming/Nabla/blob/c6d0dde3f7b5e4bde019290562997c8dfad28268/include/nbl/core/alloc/SimpleBlockBasedAllocator.h#L196

just wondering what fk-ed up along the way, cause this seems like an important bug to diagnose.